src/axl.c

```diff
@@ -309,6 +310,124 @@ int AXL_Finalize (void)
     return rc;
 }
 
+
+/** Set a AXL config parameters */
+kvtree* AXL_Config(const kvtree* config)
+{
+  kvtree* retval = (kvtree*)(config);
+
+  static int configured = 0;
+  static const char* known_options[] = {
+    AXL_KEY_CONFIG_FLUSH_ASYNC_BW,
+    AXL_KEY_CONFIG_FLUSH_ASYNC_PERCENT,
+    AXL_KEY_CONFIG_FILE_BUF_SIZE,
+    AXL_KEY_CONFIG_CRC_ON_FLUSH,
+    AXL_KEY_CONFIG_DEBUG,
+    AXL_KEY_CONFIG_MKDIR,
+    AXL_KEY_CONFIG_COPY_METADATA,
+    NULL
+  };
+
+  /* TODO: implement getting configuration options back */
+  if (config == NULL) {
+    return NULL;
+  }
+
+  if (! configured) {
+    if (config != NULL) {
+      /* dummy variables for options not actually supported (anymore) */
+      double axl_flush_async_bw = 0.0;
+      double axl_flush_async_percent = 0.0;
+      int axl_crc_on_flush = 1;
+
+      /* read out all options we know about */
+      /* TODO: this could be turned into a list of structs */
+      unsigned long ul;
+      if (kvtree_util_get_bytecount(config,
+        AXL_KEY_CONFIG_FLUSH_ASYNC_BW, &ul) == KVTREE_SUCCESS)
+      {
+        axl_flush_async_bw = (double) ul;
+        if (axl_flush_async_bw != ul) {
+          char* value;
+          kvtree_util_get_str(config, AXL_KEY_CONFIG_FLUSH_ASYNC_BW, &value);
+          AXL_ERR("Value '%s' passed for %s exceeds int range",
+            value, AXL_KEY_CONFIG_FLUSH_ASYNC_BW
+          );
+          retval = NULL;
+        }
+      }
+
+      kvtree_util_get_double(config,
+        AXL_KEY_CONFIG_FLUSH_ASYNC_PERCENT, &axl_flush_async_percent);
+
+      if (kvtree_util_get_bytecount(config,
+        AXL_KEY_CONFIG_FILE_BUF_SIZE, &ul) == KVTREE_SUCCESS)
+      {
+        axl_file_buf_size = (int) ul;
```

Comment:
`axl_file_buf_size` is a `size_t` (which most likely is an `unsigned long` on 64-bit), so you can remove the check against the `int`.  Or if you're being super-safe, check `ul` against `SIZE_MAX`.

Reply:
sorry, I had left the fix in rhaas/keyvalue_get. Cherry picked into rhaas/keyvalue. I'd rather not make assumptions about what `size_t` is wrt to `unsigned long`. The smallest type to be guaranteed by C to be 64bit is `long long int` with `long` being allowed to be 32bit.

Reply:
Then I would just make `axl_file_buf_size` an `unsigned long` and remove the check.

Reply:
I was reluctant to change types of existing variables given that I do now know here they are used. If you think this is safe, fine with me for sure (makes for easier code). Note that right now `kvtree_util_get_bytecount` and `scr_abtoull` which SCR (the original code) uses to convert byte count strings to integer variables are inconsistent. The former using `unsigned long` (which is only guaranteed to be 32bit and is indeed so even on 64bit windows machines) while the latter uses `unsigned long long` (which is the smallest type guaranteed to be able to hold a 64bit number). So unless those two are changed there will still be a need of checks (though it could / should actually go into `kvtree_util_get_bytecount` making `kvtree`'s user code much simpler).